shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/transaction/TransactionBackendHandler.java

```diff
@@ -45,6 +47,9 @@
     public TransactionBackendHandler(final TCLStatement tclStatement, final TransactionOperationType operationType, final BackendConnection backendConnection) {
         this.tclStatement = tclStatement;
         this.operationType = operationType;
+        if (tclStatement instanceof MySQLXABeginStatement) {
```

Comment:
Could we consider setting the transaction type in configuration rules? Otherwise, the actual transaction type may be inconsistent with the rules.
```
rules:
  - !TRANSACTION
    defaultType: XA
    providerType: Atomikos
```

Reply:
well, I think non distribution transaction should be LOCAL, and Atomikos or other provider is only for XA transaction.